### DISCUSSION TITLE
Using existing VPC

### DISCUSSION THREAD

**Discussion by julianblue** [developmentseed/eoapi-cdk#101] [Q&A]
Hi, 

I am trying to deploy the stack using an existing vpc. The thought of line was to replace 

https://github.com/developmentseed/eoapi-cdk/blob/main/integration_tests/cdk/app.py#L27-L35

and referencing the existing vpc by its id. I am facing issues getting this running. Are there any insights on how to do this ?

The aim is to not have publicly accessible endpoints, but internally accessible within the vpc. 

Thanks in advance.

**Comment by frederichaase** [comment]
+1 This would be really useful!

**Comment by emileten** [comment]
Thanks for the discussion @julianblue !

Could you be more specific regarding this statement 

> I am facing issues getting this running.

What did you try exactly and what error did you get?

There is a well documented method to import an already existing VPC in a given stack, [Vpc.from_lookup](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html#static-fromwbrlookupscope-id-options). Perhaps you have already tried without success?

**Reply by julianblue** [reply]
Hi @emileten ,
thanks for the quick reply. Additional changes such as vpc_subnet selection seemed to have been needed and we could not fully get it to run as expected. Is there an example app.py that uses an existing vpc and uses a custom domain name for the separate components (very similar to the eoapi dev deployed, but not publicly accessible) ? 
Appreciate any insights here. Thanks.

**Comment by julianblue** [comment]
Hi @emileten , 
any updates here ? There seem to be some processes in the cdc that hinder the use of an existing vpc. 
I appreciate any input. 
Thanks

**Reply by emileten** [reply]
Hi @julianblue, 

> Is there an example app.py that uses an existing vpc

No, there is no example of `eoapi-cdk` constructs usage we have internally that makes use of an imported vpc. In the few example applications we have, we always create a new vpc. 

That being said, we have an application that does _not_ make use of the `eoapi-cdk` constructs but deploys the same infrastructure for the VEDA project (it just doesn't import the constructs and re-implements them within the app) . [They import an existing VPC using the `Vpc.from_lookup` method that I mentioned above.](https://github.com/NASA-IMPACT/veda-backend/blob/d30bc5944dd4a350089cf6caae92e2a3e952f5fe/network/infrastructure/construct.py#L30). 

> Additional changes such as vpc_subnet selection seemed to have been needed and we could not fully get it to run as expected. 

Do you have a code snippet to show and perhaps the corresponding error ? I can take a quick look. I unfortunately won't have time to test things myself as I am caught up in other projects.

**Reply by julianblue** [reply]
Thanks @emileten . That helped. 
As the cdk creates an http api-gateway, does this mean the endpoints currently can only be public ? Or am I missing something.
